docs/reference/secrets-reference.md

```diff
@@ -0,0 +1,42 @@
+---
+id: secrets-reference
+title: Secrets YAML Reference
+---
+Recyclarr supports secret value substitution within it's YAML configuration files. This keeps sensitive
+values out of configuration files and makes it easier to share configurations without having to
+perform manual redaction. Using secrets is optional and support was introduced in version `TBD`.
+
+The `secrets.yml` file is located in the following locations (by default) depending on your
+platform:
+
+| Platform | Location                                               |
+| -------- | ------------------------------------------------------ |
+| Windows  | `%APPDATA%\recyclarr\secrets.yml`                     |
+| Linux    | `~/.config/recyclarr/secrets.yml`                     |
+| MacOS    | `~/Library/Application Support/recyclarr/secrets.yml` |
+
+Secrets can be used in both the Recyclarr config and settings files. Only one secrets file is supported.
+
+This file is optional and does not need to exist in order for Recyclarr to work properly.
+
+## Secrets Definition
+
+Define secrets in `secrets.yml` using key/value pairs. Keys must be unique.
+
+```yml
+sonarr_baseurl: http://localhost:8989
+sonarr_apikey: f7e74ba6c80046e39e076a27af5a8444
+radarr4k_url: http://localhost:7878
+radarr4k_apikey: bf99da49d0b0488ea34e4464aa63a0e5
+```
+
+## Using Secrets
+
+Substitute secrets within Recyclarr configuratopms or settings files using `!secret <key_name>`. The
+configuration will fail to load if there are references to keys that don't exist in the `secrets.yml`.
+
+```yml
+radarr:
+  - base_url: !secret radarr4k_url
+    api_key: !secret radarr4k_apikey
+```
```

Comment:
Hard to tell from the diff if there's a trailing newline at the end of the file. If not, would you mind adding one? It's the cherry on top :-)